src/index.ml

```diff
@@ -214,19 +226,22 @@ module Make (K : Key) (V : Value) (IO : IO) = struct
           let highest_entry = get_entry_iff_needed t i high highest_entry in
           let lowest_hash = float_of_int (K.hash lowest_entry.key) in
           let highest_hash = float_of_int (K.hash highest_entry.key) in
-          if lowest_hash > hashed_key || highest_hash < hashed_key then None
+          if lowest_hash > hashed_keyf || highest_hash < hashed_keyf then None
           else
             let doff =
               floor
                 ( (high -. low)
-                *. (hashed_key -. lowest_hash)
+                *. (hashed_keyf -. lowest_hash)
                 /. (highest_hash -. lowest_hash) )
             in
             let off = low +. doff -. mod_float doff entry_sizef in
             let offL = Int64.of_float off in
             let e = get_entry t i offL in
+            let hashed_e = K.hash e.key in
             if K.equal e.key key then Some e.value
```

Comment:
You could perform `K.equal` only when the hashes are equal. `hashed_e` must be computed for every iterations except the last one, anyway.